SmartDeviceLink/SDLLifecycleManager.m

```diff
@@ -246,6 +250,45 @@ - (void)didEnterStateConnected {
 }
 
 - (void)didEnterStateRegistered {
+    NSArray<SDLLanguage> *supportedLanguages = self.configuration.lifecycleConfig.languagesSupported;
+    SDLLanguage desiredLanguage = self.configuration.lifecycleConfig.language;
+    SDLLanguage actualLanguage = self.registerResponse.language;
+    BOOL delegateCanUpdateLifecycle = [self.delegate respondsToSelector:@selector(managerWillUpdateLifecycleToLanguage:)];
+    
+    // language mismatch? but actual language is a supported language? and delegate has implemented method?
+    if (actualLanguage != desiredLanguage && [supportedLanguages containsObject:actualLanguage] && delegateCanUpdateLifecycle) {
+        [self.lifecycleStateMachine transitionToState:SDLLifecycleStateUpdatingConfiguration];
+    } else {
+        [self.lifecycleStateMachine transitionToState:SDLLifecycleStateSettingUpManagers];
+    }
+}
+
+- (void)didEnterStateUpdatingConfiguration {
+    // we can expect that the delegate has implemented the update method and the actual language is a supported language
+    SDLLanguage actualLanguage = self.registerResponse.language;
+    // ask the delegate for update-config object
+    SDLLifecycleConfigurationUpdate *configUpdate = [self.delegate managerWillUpdateLifecycleToLanguage:actualLanguage];
+    
+    if (configUpdate) {
+        self.configuration.lifecycleConfig.language = actualLanguage;
+        self.configuration.lifecycleConfig.appName = configUpdate.appName;
+        self.configuration.lifecycleConfig.shortAppName = configUpdate.shortAppName;
+        self.configuration.lifecycleConfig.ttsName = configUpdate.ttsName;
+        self.configuration.lifecycleConfig.voiceRecognitionCommandNames = configUpdate.voiceRecognitionCommandNames;
```

Comment:
This doesn't quite look right to me, shouldn't it be checking if the config update is nil and not resetting the lifecycleconfig if it is?